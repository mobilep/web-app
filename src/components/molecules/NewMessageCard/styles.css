.NewMessageCard::before {
	content: '';
	position: absolute;
	border-bottom: 2px solid var(--grey2);
	width: calc(100% * .72);
	height: 100%;
	left: calc((100% - 100% * .72) / 2);
}

.NewMessageCard-first::before {
	border-top: 2px solid var(--grey2);
}

@media (max-width: 768px) {
	.NewMessageCard {
		height: 70px;
		background-color: var(--white);
	}

	.NewMessageCard::before {
		width: calc(100% - 75px);
		left: 75px;
		border-bottom: 1px solid var(--grey2);
	}

	.NewMessageCard-first::before {
		border-top: none;
	}
}
