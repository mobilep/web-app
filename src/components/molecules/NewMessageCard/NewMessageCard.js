import React from 'react';
import PropTypes from 'prop-types';

import UserCard from '../UserCard';
import './styles.css';

const NewMessageCard = ({ active, avatar, avatarColor, person, personInitials, first }) => (
	<UserCard {...{ active,
		avatar,
		avatarColor,
		person,
		personInitials,
		type: UserCard.CARD_TYPES.NEW_MESSAGE_CARD,
		className: `NewMessageCard ${first ? 'NewMessageCard-first' : ''}` } } />
);

NewMessageCard.propTypes = {
	active: PropTypes.bool,
	avatar: PropTypes.string,
	avatarColor: PropTypes.string,
	first: PropTypes.bool,
	person: PropTypes.string,
	personInitials: PropTypes.string,
};

export default NewMessageCard;
