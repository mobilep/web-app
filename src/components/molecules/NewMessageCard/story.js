import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';

import NewMessageCard from './NewMessageCard';

storiesOf('2. Molecules / NewMessageCard', module)
	.add('with text', () => (
		<NewMessageCard
			personInitials={text('Person initials', 'al')}
			avatar={text('Person avatar', 'https://i.ytimg.com/vi/tESeU278FpU/maxresdefault.jpg')}
			person={text('Person name', 'Alvin Lane')}
			active={boolean('Active', false)}
		/>
	));
