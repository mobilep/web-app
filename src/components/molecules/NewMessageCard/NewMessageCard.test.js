import React from 'react';
import { shallow } from 'enzyme';

import NewMessageCard from './NewMessageCard';

const PROPS = {
	personInitials: 'AL',
	avatar: 'https://i.ytimg.com/vi/tESeU278FpU/maxresdefault.jpg',
	person: 'Alvin Lane',
};

describe('Molecule - NewMessageCard', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<NewMessageCard
				{...PROPS}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render in active state', () => {
		const wrapper = shallow(
			<NewMessageCard
				{...PROPS}
				active
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
