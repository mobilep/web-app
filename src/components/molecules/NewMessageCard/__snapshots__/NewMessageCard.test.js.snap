// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - NewMessageCard Should render 1`] = `
<Wrapper
  avatar="https://i.ytimg.com/vi/tESeU278FpU/maxresdefault.jpg"
  className="NewMessageCard "
  person="Alvin Lane"
  personInitials="AL"
  type="NEW_MESSAGE_CARD"
/>
`;

exports[`Molecule - NewMessageCard Should render in active state 1`] = `
<Wrapper
  active={true}
  avatar="https://i.ytimg.com/vi/tESeU278FpU/maxresdefault.jpg"
  className="NewMessageCard "
  person="Alvin Lane"
  personInitials="AL"
  type="NEW_MESSAGE_CARD"
/>
`;
